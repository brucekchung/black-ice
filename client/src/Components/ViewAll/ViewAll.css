.ViewAll {
  color: #eef0f2;
  font-weight: 300;
  position: relative;
  box-sizing: border-box;
}

.ViewAll .search-bar {
  border: 1px solid #eef0f2;
  position: absolute;
  right: 24px;
  top: -75px;
}

.ViewAll h2 {
  font-size: 2rem;
  margin: 40px;
  text-align: center;
}

.ViewAll table {
  margin: 100px auto 0 auto;
  width: 90%;
}

.ViewAll button {
  width: 100%;
}

.ViewAll table thead {
  border-bottom: 1px solid #eef0f2;
}

.ViewAll table thead th {
  padding-bottom: 20px;
  font-weight: 400;
}

.ViewAll table tbody td,
.ViewAll table thead th {
  text-align: left;
  padding: 10px;
  min-width: 70px;
}

.ViewAll input:focus {
  border: 2px solid #0892a5;
}

.editable-true {
  border: 1px solid #eef0f2;
  outline: none;
}

.editable-true:focus {
  border: 2px solid #0892a5;
}