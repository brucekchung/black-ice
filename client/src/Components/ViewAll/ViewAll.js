import React, { Component } from 'react'
import Nav from '../Nav/Nav'
import { apiCall } from '../../apiCall/apiCall'
import { shape, array } from 'prop-types'

class ViewAll extends Component {
  constructor() {
    super()
    this.state = {
      data: [],
      filteredData: [],
      editableContent: null,
      hasSearched: false
    }
  }

  componentDidMount = async () => {
    const samples = await apiCall('/api/v1/samples')
    const locations = await apiCall('/api/v1/locations')
    const data = samples.map(sample => {
      const sample_id = {sample_id: sample.id}
      const found = locations.find(location => location.id === sample.locations_id)
      return Object.assign(sample, sample_id, found )
    })

    this.setState({ data })
  }

  editData = button => {
    const id = button.parentNode.parentNode.id

    button.innerText = 'Save'
    this.setState({ editableContent: id })
  }

  getRowData = button => {
    const tableRow = button.parentNode.parentNode
    const id = tableRow.id
    const editedRow = [...tableRow.childNodes].splice(0,10)
    const match = this.state.data.find(dataPoint => dataPoint.id === parseInt(id, 10))

    editedRow.forEach(tableData => match[tableData.className] = tableData.innerText)
    return match
  }

  updateData = async data => {
    const updatedData = {
      reflectance: data.reflectance,
      wavelength: data.wavelength
    }

    const url = `/api/v1/samples/${ data.sample_id }`
    const init = {
      method: 'PUT',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(updatedData)
    }

    apiCall(url, init)
  }

  saveData = async button => {
    const data = this.getRowData(button)

    await this.updateData(data)
    button.innerText = 'Edit'
    this.setState({ editableContent: null })
  }

  handleEdit = e => {
    const button = e.target

    button.innerText === 'Edit' ? this.editData(button) : this.saveData(button)
  }

  deleteRow = async e => {
    const tableRow = e.target.parentNode.parentNode
    const id = parseInt(tableRow.id, 10)
    const remainingData = this.state.data.filter(dataPoint => dataPoint.sample_id !== id)
    const url = `/api/v1/samples/${ id }`
    const init = { method: 'DELETE' }

    await apiCall(url, init)

    this.setState({ data: remainingData })
  }

  renderData = () => {
    const data = this.state.hasSearched ? this.state.filteredData : this.state.data

    return data.map(dataPoint => {
      const editable = parseInt(this.state.editableContent, 10) === dataPoint.id

      return (
        <tr id={ dataPoint.sample_id } key={ dataPoint.sample_id }>
          <td className='name'>{ dataPoint.name }</td>
          <td className='date_collected'>{ dataPoint.date_collected }</td>
          <td className='alt'>{ dataPoint.alt }</td>
          <td className='lat'>{ dataPoint.lat }</td>
          <td className='lng'>{ dataPoint.lng }</td>
          <td className='region'>{ dataPoint.region}</td>
          <td className='country'>{ dataPoint.country }</td>
          <td className='reflectance' contentEditable={ editable }>{ dataPoint.reflectance }</td>
          <td className='wavelength' contentEditable={ editable }>{ dataPoint.wavelength }</td>
          <td className='sample_id'>{ dataPoint.sample_id }</td> 
          <td><button onClick={ this.handleEdit }>Edit</button></td>
          <td><button className='delete-btn' onClick={ this.deleteRow }>Delete</button></td>
        </tr>
      )
    })
  }

  filterData = searchValue => {
    return this.state.data.filter(dataPoint => {
      const values = Object.values(dataPoint)
      const match = values.find(value => {
        let stuff = value

        if(typeof value === 'number') {
          stuff = value.toString()
        }

        return stuff.includes(searchValue)
      }) 

      return match
    })
  }

  handleChange = e => {
    const searchValue = e.target.value
    const filteredData = this.filterData(searchValue)

    this.setState({ 
      filteredData,
      hasSearched: true 
    })
  }

  render() {
    return (
      <div>
        <Nav />
        <h2>All Data</h2>
        <input type="text" name="search" onChange={ this.handleChange } /> 
        <table>
          <thead>
            <tr>
              <th>Name</th>
              <th>Date Collected</th>
              <th>Altitude</th>
              <th>Latitude</th>
              <th>Longitude</th>
              <th>Region</th>
              <th>Country</th>
              <th>Reflectance</th>
              <th>Wavelength</th>
              <th>Sample ID</th>
              <th></th>
              <th></th>
            </tr>
          </thead>
          <tbody>
            { this.renderData() }
          </tbody>
        </table>
      </div>
    )
  }
}

ViewAll.propTypes = {
  state: shape({
    data: array.isRequired,
    filteredData: array.isRequired,
  })
}

export default ViewAll
