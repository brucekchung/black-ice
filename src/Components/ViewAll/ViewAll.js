import React, { Component } from 'react'
import Nav from '../Nav/Nav'

class ViewAll extends Component {
  constructor() {
    super()
    this.state = {
      data: [{
        id: '1',
        name: 'Col',
        date_collected: '08/19/18',
        reflectance: '89%',
        wavelength: '20um',
        altitude: '4000m',
        latlng: '1, -1',
        country: 'test'
      }, {
        id: '2',
        name: 'Base',
        date_collected: '08/18/18',
        reflectance: '70%',
        wavelength: '110um',
        altitude: '5000m',
        latlng: '2, -2',
        country: 'test2'       
      }],
      search: '',
      filteredData: []
    }
  }

  // componentDidMount = async () => {
  //   const url = '/api/v1/data'
  //   const data = await apiCall(url)

  //   this.setState({ data })
  // }

  renderData = () => {
    const data = this.state.filteredData.length > 0 ? this.state.filteredData : this.state.data

    return data.map(dataPoint => {
      return (
        <tr id={ dataPoint.id } key={ dataPoint.id }>
          <td>{ dataPoint.name }</td>
          <td>{ dataPoint.date_collected }</td>
          <td>{ dataPoint.reflectance }</td>
          <td>{ dataPoint.wavelength }</td>
          <td>{ dataPoint.altitude }</td>
          <td>{ dataPoint.latlng }</td>
          <td>{ dataPoint.country }</td>
          <td>Edit</td>
          <td>Delete</td>
        </tr>
      )
    })
  }

  filterData = searchValue => {
    return this.state.data.filter(dataPoint => {
      const values = Object.values(dataPoint)

      return values.find(value => value.includes(searchValue))
    })
  }

  handleChange = e => {
    const searchValue = e.target.value
    const filteredData = this.filterData(searchValue)

    this.setState({ filteredData })
  }

  render() {
    return (
      <div>
        <Nav />
        <h2>All Data</h2>
        <input type="text" name="search" onChange={ this.handleChange } /> 
        <table>
          <thead>
            <tr>
              <th>Name</th>
              <th>Date Collected</th>
              <th>Reflectance</th>
              <th>Wavelength</th>
              <th>Altitude</th>
              <th>Lat/Lng</th>
              <th>Country</th>
              <th></th>
              <th></th>
            </tr>
          </thead>
          <tbody>
            { this.renderData() }
          </tbody>
        </table>
      </div>
    )
  }
}

export default ViewAll
