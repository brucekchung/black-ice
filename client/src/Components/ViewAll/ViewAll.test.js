import React from 'react'
import ViewAll from './ViewAll'
import { shallow } from 'enzyme'

describe('ViewAll', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<ViewAll />, { disableLifecycleMethods: true })
  })

  it('should match snapshot upon load', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should have a default state', () => {
    const expected = {
      data: [],
      filteredData: [],
      editableContent: null
    }
    const actual = wrapper.state()

    expect(actual).toEqual(expected)
  })

  it('should make an apiCall on componentDidMount', () => {

  })

  it('should setState on componentDidMount', () => {

  })

  it('should setState when editData button is clicked', () => {
    // const e = {
    //   target: {
    //     innerText: 'Add Data'
    //   },
    //   preventDefault: jest.fn()
    // }

    expect(wrapper.state('editableContent')).toEqual(null)
    // console.log(wrapper.debug())
    // wrapper.instance().handleClick(e)
    // expect(wrapper.state('data').length).toEqual(2)

  })

  it('should make an apiCall when updateData is run', () => {

  })

  it('should setState when saveData is called', () => {

  })

  it('should setState when deleteRow is called', () => {

  })

  it('should setState with the filtered data on handleChange', () => {

  })

})
