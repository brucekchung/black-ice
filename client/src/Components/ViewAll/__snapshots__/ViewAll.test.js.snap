// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewAll should match snapshot upon load 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ViewAll />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Nav />,
        <h2>
          All Data
</h2>,
        <input
          name="search"
          onChange={[Function]}
          type="text"
/>,
        <table>
          <thead>
                    <tr>
                              <th>
                                        Name
                              </th>
                              <th>
                                        Date Collected
                              </th>
                              <th>
                                        Altitude
                              </th>
                              <th>
                                        Latitude
                              </th>
                              <th>
                                        Longitude
                              </th>
                              <th>
                                        Region
                              </th>
                              <th>
                                        Country
                              </th>
                              <th>
                                        Reflectance
                              </th>
                              <th>
                                        Wavelength
                              </th>
                              <th>
                                        Sample ID
                              </th>
                              <th />
                              <th />
                    </tr>
          </thead>
          <tbody>
                    
          </tbody>
</table>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "All Data",
        },
        "ref": null,
        "rendered": "All Data",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "search",
          "onChange": [Function],
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <thead>
              <tr>
                            <th>
                                          Name
                            </th>
                            <th>
                                          Date Collected
                            </th>
                            <th>
                                          Altitude
                            </th>
                            <th>
                                          Latitude
                            </th>
                            <th>
                                          Longitude
                            </th>
                            <th>
                                          Region
                            </th>
                            <th>
                                          Country
                            </th>
                            <th>
                                          Reflectance
                            </th>
                            <th>
                                          Wavelength
                            </th>
                            <th>
                                          Sample ID
                            </th>
                            <th />
                            <th />
              </tr>
</thead>,
            <tbody>
              
</tbody>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr>
                <th>
                                Name
                </th>
                <th>
                                Date Collected
                </th>
                <th>
                                Altitude
                </th>
                <th>
                                Latitude
                </th>
                <th>
                                Longitude
                </th>
                <th>
                                Region
                </th>
                <th>
                                Country
                </th>
                <th>
                                Reflectance
                </th>
                <th>
                                Wavelength
                </th>
                <th>
                                Sample ID
                </th>
                <th />
                <th />
</tr>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <th>
                    Name
</th>,
                  <th>
                    Date Collected
</th>,
                  <th>
                    Altitude
</th>,
                  <th>
                    Latitude
</th>,
                  <th>
                    Longitude
</th>,
                  <th>
                    Region
</th>,
                  <th>
                    Country
</th>,
                  <th>
                    Reflectance
</th>,
                  <th>
                    Wavelength
</th>,
                  <th>
                    Sample ID
</th>,
                  <th />,
                  <th />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Name",
                  },
                  "ref": null,
                  "rendered": "Name",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Date Collected",
                  },
                  "ref": null,
                  "rendered": "Date Collected",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Altitude",
                  },
                  "ref": null,
                  "rendered": "Altitude",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Latitude",
                  },
                  "ref": null,
                  "rendered": "Latitude",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Longitude",
                  },
                  "ref": null,
                  "rendered": "Longitude",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Region",
                  },
                  "ref": null,
                  "rendered": "Region",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Country",
                  },
                  "ref": null,
                  "rendered": "Country",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Reflectance",
                  },
                  "ref": null,
                  "rendered": "Reflectance",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Wavelength",
                  },
                  "ref": null,
                  "rendered": "Wavelength",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Sample ID",
                  },
                  "ref": null,
                  "rendered": "Sample ID",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "th",
                },
              ],
              "type": "tr",
            },
            "type": "thead",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
            },
            "ref": null,
            "rendered": Array [],
            "type": "tbody",
          },
        ],
        "type": "table",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Nav />,
          <h2>
            All Data
</h2>,
          <input
            name="search"
            onChange={[Function]}
            type="text"
/>,
          <table>
            <thead>
                        <tr>
                                    <th>
                                                Name
                                    </th>
                                    <th>
                                                Date Collected
                                    </th>
                                    <th>
                                                Altitude
                                    </th>
                                    <th>
                                                Latitude
                                    </th>
                                    <th>
                                                Longitude
                                    </th>
                                    <th>
                                                Region
                                    </th>
                                    <th>
                                                Country
                                    </th>
                                    <th>
                                                Reflectance
                                    </th>
                                    <th>
                                                Wavelength
                                    </th>
                                    <th>
                                                Sample ID
                                    </th>
                                    <th />
                                    <th />
                        </tr>
            </thead>
            <tbody>
                        
            </tbody>
</table>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "All Data",
          },
          "ref": null,
          "rendered": "All Data",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "search",
            "onChange": [Function],
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <thead>
                <tr>
                                <th>
                                                Name
                                </th>
                                <th>
                                                Date Collected
                                </th>
                                <th>
                                                Altitude
                                </th>
                                <th>
                                                Latitude
                                </th>
                                <th>
                                                Longitude
                                </th>
                                <th>
                                                Region
                                </th>
                                <th>
                                                Country
                                </th>
                                <th>
                                                Reflectance
                                </th>
                                <th>
                                                Wavelength
                                </th>
                                <th>
                                                Sample ID
                                </th>
                                <th />
                                <th />
                </tr>
</thead>,
              <tbody>
                
</tbody>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <tr>
                  <th>
                                    Name
                  </th>
                  <th>
                                    Date Collected
                  </th>
                  <th>
                                    Altitude
                  </th>
                  <th>
                                    Latitude
                  </th>
                  <th>
                                    Longitude
                  </th>
                  <th>
                                    Region
                  </th>
                  <th>
                                    Country
                  </th>
                  <th>
                                    Reflectance
                  </th>
                  <th>
                                    Wavelength
                  </th>
                  <th>
                                    Sample ID
                  </th>
                  <th />
                  <th />
</tr>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <th>
                      Name
</th>,
                    <th>
                      Date Collected
</th>,
                    <th>
                      Altitude
</th>,
                    <th>
                      Latitude
</th>,
                    <th>
                      Longitude
</th>,
                    <th>
                      Region
</th>,
                    <th>
                      Country
</th>,
                    <th>
                      Reflectance
</th>,
                    <th>
                      Wavelength
</th>,
                    <th>
                      Sample ID
</th>,
                    <th />,
                    <th />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Name",
                    },
                    "ref": null,
                    "rendered": "Name",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Date Collected",
                    },
                    "ref": null,
                    "rendered": "Date Collected",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Altitude",
                    },
                    "ref": null,
                    "rendered": "Altitude",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Latitude",
                    },
                    "ref": null,
                    "rendered": "Latitude",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Longitude",
                    },
                    "ref": null,
                    "rendered": "Longitude",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Region",
                    },
                    "ref": null,
                    "rendered": "Region",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Country",
                    },
                    "ref": null,
                    "rendered": "Country",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Reflectance",
                    },
                    "ref": null,
                    "rendered": "Reflectance",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Wavelength",
                    },
                    "ref": null,
                    "rendered": "Wavelength",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Sample ID",
                    },
                    "ref": null,
                    "rendered": "Sample ID",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "th",
                  },
                ],
                "type": "tr",
              },
              "type": "thead",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
              },
              "ref": null,
              "rendered": Array [],
              "type": "tbody",
            },
          ],
          "type": "table",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
